aten/src/ATen/native/cuda/Copy.cu

```diff
@@ -380,8 +463,12 @@ static void copy_kernel_cuda(TensorIterator& iter, bool non_blocking) {
     // TODO: warn on the return value.
     CachingHostAllocator_recordEvent(ptr, ctx, stream);
 
-  } else {
-    at::cuda::memcpy_and_sync(dst, src, nbytes, kind, stream);
+  } else { 
```

Comment:
you are fixing only the blocking case, whereas usually non-blocking are most-performance sensitive. Please fix non-blocking case also.